src/ConfigR/BasicConfig.Dictionary.cs

```diff
@@ -57,7 +59,7 @@ public bool ContainsKey(string key)
 
         public bool Remove(string key)
         {
-            log.TraceFormat(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource());
+            log.TraceFormat(string.Format(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource()));

```

Comment:
Cool, thanks for the feedback. I will simplify the calls and make them consistent throughout.

-----Original Message-----
From: "Adam Ralph" <notifications@github.com>
Sent: ‎08/‎10/‎2015 21:15
To: "config-r/config-r" <config-r@noreply.github.com>
Cc: "Christos Matskas" <christos.matskas@softwarelounge.co.uk>
Subject: Re: [config-r] Logging uprade to LibLog (#205)

In src/ConfigR/BasicConfig.Dictionary.cs:
> @@ -57,7 +59,7 @@ public bool ContainsKey(string key)
>  
>          public bool Remove(string key)
>          {
> -            log.TraceFormat(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource());
> +            log.TraceFormat(string.Format(CultureInfo.InvariantCulture, "Removing '{0}' from {1}", key, this.GetSource()));
Sure, but if we discard CultureInfo.CurrentCulture then the call becomes:
log.TraceFormat("Removing '{0}' from {1}", key, this.GetSource());There is no redundant evaluation taking place because we are passing the arguments to LibLog and it will only use them to format a string message if the log level is enabled.
Deferred execution via lambdas is only useful when you are using some other operation to generate a string, and the cost of allocating a closure is outweighed by the cost of generating the string. I.e. you would never write
logger.Debug( () => "hello world");because the cost of allocating the closure will outweigh the cost of embedding the string literal "hello world" in the output assembly.
—
Reply to this email directly or view it on GitHub.